psi4/src/psi4/cchbar/Wabei_BBBB_UHF.cc

```diff
@@ -59,198 +60,191 @@ namespace psi { namespace cchbar {
+      for(int ei=0; ei< W.params->rowtot[Gei]; ei++){
+         global_dpd_->buf4_mat_irrep_row_rd(&W, Gei, ei);
+         C_DGEMV('t',nrows,ncols,-1,T.matrix[Gei][0],ncols,Z.matrix[Gei][ei],1,
+             1.0,W.matrix[Gei][0],1);
+         global_dpd_->buf4_mat_irrep_row_wrt(&W, Gei, ei);
+      }
+      global_dpd_->buf4_mat_irrep_row_close(&W, Gei);
+      global_dpd_->buf4_mat_irrep_close(&T, Gnm);
+      global_dpd_->buf4_mat_irrep_close(&Z, Gei);
+    }
+  }
+  global_dpd_->buf4_close(&T);
   global_dpd_->buf4_close(&Z);
-
-  /**** Term VI ****/
-
-  /** tau_mn^ab <mn||ei> --> W'(ab,ei) **/
-  global_dpd_->buf4_init(&W, PSIF_CC_TMP0, 0, 17, 31, 17, 31, 0, "W'(ab,ei)");
-  global_dpd_->buf4_init(&T2, PSIF_CC_TAMPS, 0, 12, 17, 12, 17, 0, "tauijab");
-  global_dpd_->buf4_init(&E, PSIF_CC_EINTS, 0, 12, 31, 12, 31, 0, "E <ij||ka> (i>j,ak)");
-  global_dpd_->contract444(&T2, &E, &W, 1, 1, -1, 1);
-  global_dpd_->buf4_close(&E);
-  global_dpd_->buf4_close(&T2);
   global_dpd_->buf4_close(&W);
 
-  /**** Term VII ****/
-
-  /** Wabei <-- <bm||ef> t_im^af + <bM|eF> t_iM^aF - <am||ef> t_im^bf - <aM|eF> t_iM^bF
-      Evaluate in six steps:
-        (1) Sort <bm||ef> and <bM|eF> to F(be,mf) and F(be,MF) ordering.
-        (2) Z(be,ia) = F(be,mf) T(ia,mf) + F(be,MF) T(ia,MF)
-        (3) Sort Z(be,ia) --> Z'(ei,ab)
-	(4) Sort Z'(ei,ab) --> Z''(ei,ba)
-        (5) AXPY: Z'(ei,ab) = Z'(ei,ab) - Z''(ei,ba)
-        (6) AXPY: W(ei,ab) <-- Z'(ei,ab)
-      NB: The storage for the sorts is expensive and will eventually require out-of-core
-          codes.
-  **/
-
-  /** <bm||ef> --> F(be,mf) **/
-  global_dpd_->buf4_init(&F, PSIF_CC_FINTS, 0, 31, 15, 31, 15, 1, "F <ai|bc>");
-  global_dpd_->buf4_sort(&F, PSIF_CC_FINTS, prqs, 15, 30, "F <ai||bc> (ab,ic)");
-  global_dpd_->buf4_close(&F);
-
-  /** <bM|eF> --> (be,MF) **/
-  global_dpd_->buf4_init(&F, PSIF_CC_FINTS, 0, 25, 29, 25, 29, 0, "F <aI|bC>");
-  global_dpd_->buf4_sort(&F, PSIF_CC_FINTS, prqs, 15, 20, "F <aI|bC> (ab,IC)");
-  global_dpd_->buf4_close(&F);
-
-  /** <bm||ef> t_im^af --> Z(be,ia) **/
-  global_dpd_->buf4_init(&Z, PSIF_CC_TMP0, 0, 15, 30, 15, 30, 0, "Z(be,ia)");
+  /*
+   * <ai||bc> sorted F(ab,ic)
+   * Build Z1(ia,mf) = t_{im}^{af} -t_i^ft_m^a
+   * contract F(ab,ic)Z1(ia,mf) => W1(be,ia)
+   * <aI|bC> sorted F(ab,IC)
+   * t_iM^aF = t(ia,MF)
+   * contract F(ab,IC)t(ia,MF) => W1(be,ia)
+   * sort W1(be,ia) W2(ei,ab)
+   * Read W2 with anti=1
+   * Axpy W2=> Weiab
+   *
+   * --AMJ 02/19/2016
+   **/
+  build_Z1_BBBB();//Z(ia,mf)
+  global_dpd_->buf4_init(&Z, PSIF_CC_TMP0, 0, 30, 30, 30, 30, 0, "Z1(ia,mf)");
   global_dpd_->buf4_init(&F, PSIF_CC_FINTS, 0, 15, 30, 15, 30, 0, "F <ai||bc> (ab,ic)");
-  global_dpd_->buf4_init(&T2, PSIF_CC_TAMPS, 0, 30, 30, 30, 30, 0, "tiajb");
-  global_dpd_->contract444(&F, &T2, &Z, 0, 0, -1, 0);
-  global_dpd_->buf4_close(&T2);
+  global_dpd_->buf4_init(&W, PSIF_CC_TMP0, 0, 15, 30, 15, 30,  0, "W1(be,ia)");
```

Comment:
Are items like this changing the identity of psi scratch files, such that the comments in psifiles.h need editing? Or just labels internal to DPD?